diagnostic_feedback/public/js/src/student/student_quiz.js

```diff
@@ -117,7 +117,16 @@ function StudentQuiz(runtime, element, initData) {
       return {'question_id': id, 'student_choice': studentChoice};
     }
 
-    function submitQuestionResponse(isLast, currentStep) {
+    function getQuestionEventData(choice) {
+      // get text of question and selected answer text
+      var choice = $('input[type="radio"][value="' + choice + '"]:visible'),
+        student_choice = choice.next().text(),
+        question_txt = choice.parent().prev().find('p').text();
```

Comment:
@jareerahsan The code in the three lines above seems a bit brittle: For example, the logic for obtaining `question_txt` would break when introducing another element between the parent of `choice` and the element that contains the question text. You could make this code a bit more robust by relying more on CSS classes (and less on relative positions of elements) to find the elements that you are looking for. To give a concrete example of what I mean, you could do this to obtain the `question_txt`:

```js
question_txt = choice.parents('.q-container').find('.quiz-question').text();
```

Similarly, to find the `student_choice` you could do something like this:

```js
var choice_id = choice.attr('id'),
  student_choice = choice.parent('.user-answers').find('label[for="' + choice_id + '"]');
```

---

Lastly, if you are using a unique `value` for each of the answer choices, you should be able to remove the `:visible` portion from the selector for `choice`.